### ISSUE TITLE
Crash when using keyboard on inputText

### ISSUE THREAD

**Issue by freeboub** [react-native-tvos/react-native-tvos#279]
## Description

When we are using Input on Android 11 device in some case we generate following crash:

03-15 13:13:19.923 12015 12015 D AndroidRuntime: Shutting down VM
03-15 13:13:19.927 12015 12015 E AndroidRuntime: FATAL EXCEPTION: main
03-15 13:13:19.927 12015 12015 E AndroidRuntime: Process: com.starhub.tv.stb, PID: 12015
03-15 13:13:19.927 12015 12015 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'int android.view.View.getId()' on a null object reference
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at com.facebook.react.views.textinput.ReactTextInputManager$4.onEditorAction(ReactTextInputManager.java:1097)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at android.widget.TextView.onEditorAction(TextView.java:7058)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at com.android.internal.widget.EditableInputConnection.performEditorAction(EditableInputConnection.java:138)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:363)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:93)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:223)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7656)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
03-15 13:13:19.927 12015 12015 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

## React Native version:
0.66.3-0

## Steps To Reproduce
Create a screen with a simple Input field 

```
            <TextInput
                ref={forwardedRef}
                maxFontSizeMultiplier={0}
                placeholderTextColor={placeholderTextColor}
                secureTextEntry={secureTextEntry ? !showPassword : false}
                style={[styles.textInput, editable === false && styles.textInputDisabled]}
                onBlur={_onBlur}
                onFocus={_onFocus}
                {...otherProps}
            />
```

## Expected Results
No crash

## Snack, code example, screenshot, or link to a repository:
in function: onEditorAction

https://github.com/react-native-tvos/react-native-tvos/blob/740fecdaef5ebc3460e312685228f6501efe53db/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java#L1091

Line 1091, there is test  if (view != null) { 
And 1097, view is used without testing its value. This is causing the crash!

### workaround

```
            <TextInput
                ref={forwardedRef}
                maxFontSizeMultiplier={0}
                placeholderTextColor={placeholderTextColor}
                secureTextEntry={secureTextEntry ? !showPassword : false}
                style={[styles.textInput, editable === false && styles.textInputDisabled]}
                onBlur={_onBlur}
                onFocus={_onFocus}
                **nextFocusDown={forwardedRef}**
                {...otherProps}
            />
```

**Comment by douglowder** [comment]
Correction -- the issue you are seeing has a different stack trace. I'll have a look and implement a fix.  Seems like this might affect RN core so the fix should be merged there as well.

**Comment by freeboub** [comment]
@douglowder I already check react-native code, I can confirm this is react-native-tvos only.
I was not able to reproduce the issue on android 10, I don't really understand why ...
 
A trivial fix would be to move this code:

```
                    if (editText.getId() == view.getId()) {
                      editText.onFocusChanged(false, View.FOCUSABLES_ALL, null);
                    }
```

Inside the if(view != null), but I am not sure of possible side effects.
I didn't build react-native from source yet, then I cannot try a fix for now...

Thank you

**Comment by freeboub** [comment]
minor, but notice that is line is useless:
https://github.com/react-native-tvos/react-native-tvos/blob/740fecdaef5ebc3460e312685228f6501efe53db/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java#L1080

**Comment by douglowder** [comment]
Thanks @freeboub this does seem to be a regression due to the fix in https://github.com/react-native-tvos/react-native-tvos/commit/8dcf4dd0eedcceab6eaa9981d776654fd1c71733 . Agreed that line 1080 is not useful. This is a merge of a fix from an earlier RN version so maybe the merge had issues. I'll take a look.

**Comment by douglowder** [comment]
Fixed in release 0.66.3-3

**Comment by KishoreKumarNS47** [comment]
hi @douglowder 
As I see the react-native-tvos is using some aar of rect-native and we are using react-native-tvos 0.64.3 and the above issue is happening in the 0.64.3 version .So the question is how can I patch the fix to 0.64.3 as we can't upgrade the react-native-tvos to 0.66.3 due to some project specifications?
